### PR TITLE
Only install suse-openstack-cloud-release package on SLES 12.1

### PR DESCRIPTION
**Pull request by dirkmueller**
In case we need to install a SLES 12.0 node (for SES2 :-(), don't
try to install additional software on it.

### CODE REVIEW

**Comment by toabctl** [comment]
+1

File: chef/cookbooks/provisioner/templates/default/autoyast.xml.erb [hunk 1]
```diff
@@ -278,10 +278,11 @@ sync
 <% else -%>
   <% if @target_platform_version == "11.3" -%>
       <package>suse-sle11-openstack-cloud-release</package>
-  <% else -%>
-      <package>suse-openstack-cloud-release</package>
   <% end -%>
+  <% if @target_platform_version == "12.1" -%>
```
Comment (rsalevsky) [inline chef/cookbooks/provisioner/templates/default/autoyast.xml.erb]: `elsif` should be better in this case

**Comment by vuntz** [comment]
Hrm, just realized: don't we have the issue of ruby2.1-rubygem-chef? (which would be a deal breaker)

**Comment by dirkmueller** [comment]
@vuntz SES2 contains chef right now (I don't know why)

File: chef/cookbooks/provisioner/templates/default/autoyast.xml.erb [hunk 2]
```diff
@@ -278,10 +278,10 @@ sync
 <% else -%>
   <% if @target_platform_version == "11.3" -%>
       <package>suse-sle11-openstack-cloud-release</package>
```
Comment (vuntz) [inline chef/cookbooks/provisioner/templates/default/autoyast.xml.erb]: Strictly speaking, we'd need the supportutils package here too (but I'd be equally fine removing 11.3 from there)

**Comment by vuntz** [comment]
Can you also update the crowbar_register template, since I expect we will have the same issue for packages (and maybe ntp)?

File: chef/cookbooks/provisioner/templates/default/autoyast.xml.erb (hunk 2 above)
Comment (dirkmueller) [inline chef/cookbooks/provisioner/templates/default/autoyast.xml.erb]: depends on if we'll ever have a 11.3 compute nodes..

**Comment by dirkmueller** [comment]
@vuntz done

File: chef/cookbooks/provisioner/templates/default/autoyast.xml.erb (hunk 2 above)
Comment (vuntz) [inline chef/cookbooks/provisioner/templates/default/autoyast.xml.erb]: Well, there's the idea of potentially getting 11.4 compute nodes later on, and having the code here would be a reminder to update it. But again, I'm fine removing it too :-)

File: chef/cookbooks/provisioner/templates/suse/crowbar_register.erb [hunk 3]
```diff
@@ -309,8 +309,8 @@ zypper --non-interactive install -t pattern $PATTERNS_INSTALL
 # Auto-agree with the license since it was already agreed on for the admin server
 <% if @target_platform_version.to_f < 12.0 -%>
 zypper --non-interactive install --auto-agree-with-licenses suse-sle11-openstack-cloud-release
-<% else -%>
-zypper --non-interactive install --auto-agree-with-licenses suse-openstack-cloud-release
+<% elsif @target_platform_version == "12.1" -%>
+zypper --non-interactive install --auto-agree-with-licenses suse-openstack-cloud-release supportutils-plugin-suse-openstack-cloud
```
Comment (vuntz) [inline chef/cookbooks/provisioner/templates/suse/crowbar_register.erb]: Hrm, you forgot to remove the package from line 302?

Comment (dirkmueller) [inline chef/cookbooks/provisioner/templates/suse/crowbar_register.erb]: yes, fixed.

**Comment by rhafer** [comment]
+1

**Pull request merged**